### PR TITLE
Added "paste" and "functions paste" modes.

### PR DESCRIPTION
**Pull request by javadba**
"Paste" mode allows entering multiline snippets at a single shot.  "Functions Paste" mode allows entering functions that are inserted into the compilation unit  after the #includes sections and before the main.  These are making my life with igcc significantly more effective.

### CODE REVIEW

**Comment by StarlitGhost** [comment]
I've been using C++11/14 Lambda functions, but this also seems like a good solution.
I'll take a more detailed look later for merging, but thanks for the contribution!

File: README.txt [hunk 1]
```diff
@@ -28,6 +28,40 @@ It is possible to include header files you need like this:
...
+g++>  }
+g++>  return ind;
+g++> }
+g++> .f
+Functions paste mode is OFF
+
+You can use '.p' to enter "Paste" mode to enter multi line snippets :
+
+g++> .p
+Paste mode is ON: Enter ".f" again to return to return to normal editing.
```
Comment (StarlitGhost) [inline README.txt]: Shouldn't this be ".p"?

File: libigcc/dot_commands.py [hunk 2]
```diff
@@ -88,8 +106,11 @@ def dot_w( runner ):
 	".l" : ( "List the code you have entered", dot_l ),
 	".L" : ( "List the whole program as given to the compiler", dot_L ),
 	".n" : ( "Clear all entered commands ('new')", dot_n ),
+	".N" : ( "Clear all entered commands including user functions ('New')", dot_N ),
```
Comment (StarlitGhost) [inline libigcc/dot_commands.py:110]: I may change .n to take arguments instead, so you can choose which categories you want to wipe (this isn't something you need to do before I accept the PR, just noting it for myself).

File: libigcc/dot_commands.py [hunk 3]
```diff
@@ -53,6 +53,24 @@ def dot_n( runner ):
...
+
+def dot_f( runner):
+	runner.functions_paste = not runner.functions_paste
+	print 'Functions paste mode is ' + ('ON: Enter \".f\" again to return to return to normal editing.' if runner.functions_paste else 'OFF') + '\n'
+	runner.paste = False
+	return False, False 
+
+def dot_p( runner):
+	runner.paste = not runner.paste
+	print 'Paste mode is ' + ('ON: Enter \".f\" again to return to return to normal editing.' if runner.paste else 'OFF') + '\n'
```
Comment (StarlitGhost) [inline libigcc/dot_commands.py]: This should say .p too, right?

**Review by StarlitGhost — CHANGES_REQUESTED** [review]
A couple of typos to fix, but good work overall 👍

**Comment by javadba** [comment]
No '.f'  toggles function paste mode    .p Toggles paste mode.   You can try it out!
  thx!

**Comment by javadba** [comment]
There is a subtle difference as well:  toggling "off" the functions paste mode allows further editing whereas toggling off paste mode causes the buffer to be executed.  It makes sense that way. I've been using both of them for doing experiments on both function bodies and  non -function snippets.

**Comment by javadba** [comment]
Hello?? I responded essentially immediately to your comments but then .. silence ..

**Comment by StarlitGhost** [comment]
> No '.f' toggles function paste mode .p Toggles paste mode.

That's exactly what I'm saying, the 2 messages I pointed out currently say that .f exits the mode entered with .p

File: README.txt (hunk 1 above)
Comment (javadba) [reply README.txt]: wow I use that all the time and did not even notice. thx.

File: libigcc/dot_commands.py (hunk 3 above)
Comment (javadba) [reply libigcc/dot_commands.py]: Apparently I borked this when adding .f    thx

File: libigcc/dot_commands.py (hunk 2 above)
Comment (javadba) [reply libigcc/dot_commands.py:110]: K: I'll add TODO for you.

**Review by StarlitGhost — APPROVED** [review]

**Pull request merged**